Use `async_retry` instead of `retry` in google/cloud/firestore_v1/async_document.py

There is a warning which appears in the `Kokoro Prerelease Dependencies` presubmit. 

```
E                   UserWarning: Using the synchronous google.api_core.retry.Retry with asynchronous calls may lead to unexpected results. Please use google.api_core.retry_async.AsyncRetry instead.
```

One potential issue is that [retry](https://github.com/googleapis/python-api-core/blob/main/google/api_core/retry.py) is used in `google/cloud/firestore_v1/async_document.py` is used instead of [async_retry](https://github.com/googleapis/python-api-core/blob/main/google/api_core/retry_async.py) in `google/cloud/firestore_v1/async_document.py`

https://github.com/googleapis/python-firestore/blob/098c8f52834c216895faca3b00dc20130410ca6f/google/cloud/firestore_v1/async_document.py#L20


```
___________________ test_collections[system-tests-named-db] ____________________

client = 
database = 'system-tests-named-db'

    @pytest.mark.parametrize("database", [None, FIRESTORE_OTHER_DB], indirect=True)
    async def test_collections(client, database):
>       collections = [x async for x in client.collections(retry=RETRIES)]

tests/system/test_system_async.py:114: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/system/test_system_async.py:114: in 
    collections = [x async for x in client.collections(retry=RETRIES)]
google/cloud/firestore_v1/async_client.py:306: in collections
    **kwargs,
google/cloud/firestore_v1/services/firestore/async_client.py:1749: in list_collection_ids
    metadata=metadata,
.nox/prerelease_deps-3-7/lib/python3.7/site-packages/google/api_core/gapic_v1/method.py:131: in __call__
    return wrapped_func(*args, **kwargs)
.nox/prerelease_deps-3-7/lib/python3.7/site-packages/google/api_core/retry.py:377: in retry_wrapped_func
    on_error=on_error,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

target = functools.partial(.error_remapped_callable at 0x7f36866069e0>, parent: "projects/...ystem-tests-named-db/documents'), ('x-goog-api-client', 'gl-python/3.7.17 grpc/1.60.0rc1 gax/2.15.0rc1 gapic/2.13.1')])
predicate = .if_exception_type_predicate at 0x7f3685e729e0>
sleep_generator = 
timeout = 60.0, on_error = None, kwargs = {}
deadline = datetime.datetime(2023, 12, 4, 20, 35, 38, 435291), last_exc = None
sleep = 0.06341161243697828
result = 

    def retry_target(
        target, predicate, sleep_generator, timeout=None, on_error=None, **kwargs
    ):
        """Call a function and retry if it fails.
    
        This is the lowest-level retry helper. Generally, you'll use the
        higher-level retry helper :class:`Retry`.
    
        Args:
            target(Callable): The function to call and retry. This must be a
                nullary function - apply arguments with `functools.partial`.
            predicate (Callable[Exception]): A callable used to determine if an
                exception raised by the target should be considered retryable.
                It should return True to retry or False otherwise.
            sleep_generator (Iterable[float]): An infinite iterator that determines
                how long to sleep between retries.
            timeout (float): How long to keep retrying the target.
            on_error (Callable[Exception]): A function to call while processing a
                retryable exception.  Any error raised by this function will *not*
                be caught.
            deadline (float): DEPRECATED: use ``timeout`` instead. For backward
                compatibility, if specified it will override ``timeout`` parameter.
    
        Returns:
            Any: the return value of the target function.
    
        Raises:
            google.api_core.RetryError: If the deadline is exceeded while retrying.
            ValueError: If the sleep generator stops yielding values.
            Exception: If the target raises a method that isn't retryable.
        """
    
        timeout = kwargs.get("deadline", timeout)
    
        if timeout is not None:
            deadline = datetime_helpers.utcnow() + datetime.timedelta(seconds=timeout)
        else:
            deadline = None
    
        last_exc = None
    
        for sleep in sleep_generator:
            try:
                result = target()
                if inspect.isawaitable(result):
>                   warnings.warn(_ASYNC_RETRY_WARNING)
E                   UserWarning: Using the synchronous google.api_core.retry.Retry with asynchronous calls may lead to unexpected results. Please use google.api_core.retry_async.AsyncRetry instead.

.nox/prerelease_deps-3-7/lib/python3.7/site-packages/google/api_core/retry.py:209: UserWarning
```

--- a/google/cloud/firestore_v1/async_aggregation.py
+++ b/google/cloud/firestore_v1/async_aggregation.py
@@ -21,7 +21,7 @@ a more common way to create an aggregation query than direct usage of the constr
 from __future__ import annotations
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from typing import List, Union, AsyncGenerator
 
@@ -46,7 +46,7 @@ class AsyncAggregationQuery(BaseAggregationQuery):
         self,
         transaction=None,
         retry: Union[
-            retries.Retry, None, gapic_v1.method._MethodDefault
+            retries.AsyncRetry, None, gapic_v1.method._MethodDefault
         ] = gapic_v1.method.DEFAULT,
         timeout: float | None = None,
     ) -> List[AggregationResult]:
@@ -80,7 +80,7 @@ class AsyncAggregationQuery(BaseAggregationQuery):
         self,
         transaction=None,
         retry: Union[
-            retries.Retry, None, gapic_v1.method._MethodDefault
+            retries.AsyncRetry, None, gapic_v1.method._MethodDefault
         ] = gapic_v1.method.DEFAULT,
         timeout: float | None = None,
     ) -> Union[AsyncGenerator[List[AggregationResult], None]]:

--- a/google/cloud/firestore_v1/async_batch.py
+++ b/google/cloud/firestore_v1/async_batch.py
@@ -16,7 +16,7 @@
 
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from google.cloud.firestore_v1.base_batch import BaseWriteBatch
 
@@ -38,7 +38,7 @@ class AsyncWriteBatch(BaseWriteBatch):
 
     async def commit(
         self,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> list:
         """Commit the changes accumulated in this batch.

--- a/google/cloud/firestore_v1/async_client.py
+++ b/google/cloud/firestore_v1/async_client.py
@@ -25,7 +25,7 @@ In the hierarchy of API concepts
 """
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from google.cloud.firestore_v1.base_client import (
     BaseClient,
@@ -228,7 +228,7 @@ class AsyncClient(BaseClient):
         references: List[AsyncDocumentReference],
         field_paths: Iterable[str] = None,
         transaction=None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[DocumentSnapshot, Any]:
         """Retrieve a batch of documents.
@@ -284,7 +284,7 @@ class AsyncClient(BaseClient):
 
     async def collections(
         self,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[AsyncCollectionReference, Any]:
         """List top-level collections of the client's database.

--- a/google/cloud/firestore_v1/async_collection.py
+++ b/google/cloud/firestore_v1/async_collection.py
@@ -15,7 +15,7 @@
 """Classes for representing collections for the Google Cloud Firestore API."""
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from google.cloud.firestore_v1.base_collection import (
     BaseCollectionReference,
@@ -85,7 +85,7 @@ class AsyncCollectionReference(BaseCollectionReference[async_query.AsyncQuery]):
         self,
         document_data: dict,
         document_id: str = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> Tuple[Any, Any]:
         """Create a document in the Firestore database with the provided data.
@@ -144,7 +144,7 @@ class AsyncCollectionReference(BaseCollectionReference[async_query.AsyncQuery]):
     async def list_documents(
         self,
         page_size: int = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[DocumentReference, None]:
         """List all subdocuments of the current collection.
@@ -177,7 +177,7 @@ class AsyncCollectionReference(BaseCollectionReference[async_query.AsyncQuery]):
     async def get(
         self,
         transaction: Transaction = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> list:
         """Read the documents in this collection.
@@ -208,7 +208,7 @@ class AsyncCollectionReference(BaseCollectionReference[async_query.AsyncQuery]):
     async def stream(
         self,
         transaction: Transaction = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncIterator[async_document.DocumentSnapshot]:
         """Read the documents in this collection.

--- a/google/cloud/firestore_v1/async_document.py
+++ b/google/cloud/firestore_v1/async_document.py
@@ -17,7 +17,7 @@ import datetime
 import logging
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 from google.cloud._helpers import _datetime_to_pb_timestamp  # type: ignore
 
 from google.cloud.firestore_v1.base_document import (
@@ -65,7 +65,7 @@ class AsyncDocumentReference(BaseDocumentReference):
     async def create(
         self,
         document_data: dict,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> write.WriteResult:
         """Create the current document in the Firestore database.
@@ -95,7 +95,7 @@ class AsyncDocumentReference(BaseDocumentReference):
         self,
         document_data: dict,
         merge: bool = False,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> write.WriteResult:
         """Replace the current document in the Firestore database.
@@ -135,7 +135,7 @@ class AsyncDocumentReference(BaseDocumentReference):
         self,
         field_updates: dict,
         option: _helpers.WriteOption = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> write.WriteResult:
         """Update an existing document in the Firestore database.
@@ -292,7 +292,7 @@ class AsyncDocumentReference(BaseDocumentReference):
     async def delete(
         self,
         option: _helpers.WriteOption = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> Timestamp:
         """Delete the current document in the Firestore database.
@@ -327,7 +327,7 @@ class AsyncDocumentReference(BaseDocumentReference):
         self,
         field_paths: Iterable[str] = None,
         transaction=None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> DocumentSnapshot:
         """Retrieve a snapshot of the current document.
@@ -395,7 +395,7 @@ class AsyncDocumentReference(BaseDocumentReference):
     async def collections(
         self,
         page_size: int = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator:
         """List subcollections of the current document.

--- a/google/cloud/firestore_v1/async_query.py
+++ b/google/cloud/firestore_v1/async_query.py
@@ -21,7 +21,7 @@ a more common way to create a query than direct usage of the constructor.
 from __future__ import annotations
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from google.cloud import firestore_v1
 from google.cloud.firestore_v1.base_query import (
@@ -172,7 +172,7 @@ class AsyncQuery(BaseQuery):
     async def get(
         self,
         transaction: Transaction = None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> list:
         """Read the documents in the collection that match this query.
@@ -267,7 +267,7 @@ class AsyncQuery(BaseQuery):
     async def stream(
         self,
         transaction=None,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[async_document.DocumentSnapshot, None]:
         """Read the documents in the collection that match this query.
@@ -380,7 +380,7 @@ class AsyncCollectionGroup(AsyncQuery, BaseCollectionGroup):
     async def get_partitions(
         self,
         partition_count,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[QueryPartition, None]:
         """Partition a query for parallelization.

--- a/google/cloud/firestore_v1/async_transaction.py
+++ b/google/cloud/firestore_v1/async_transaction.py
@@ -19,7 +19,7 @@ import asyncio
 import random
 
 from google.api_core import gapic_v1
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 
 from google.cloud.firestore_v1.base_transaction import (
     _BaseTransactional,
@@ -153,7 +153,7 @@ class AsyncTransaction(async_batch.AsyncWriteBatch, BaseTransaction):
     async def get_all(
         self,
         references: list,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[DocumentSnapshot, Any]:
         """Retrieves multiple documents from Firestore.
@@ -176,7 +176,7 @@ class AsyncTransaction(async_batch.AsyncWriteBatch, BaseTransaction):
     async def get(
         self,
         ref_or_query,
-        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
         timeout: float = None,
     ) -> AsyncGenerator[DocumentSnapshot, Any]:
         """

--- a/tests/system/test_system_async.py
+++ b/tests/system/test_system_async.py
@@ -25,7 +25,7 @@ from typing import Callable, Dict, List, Optional
 
 from google.oauth2 import service_account
 
-from google.api_core import retry as retries
+from google.api_core import retry_async as retries
 from google.api_core import exceptions as core_exceptions
 
 from google.api_core.exceptions import AlreadyExists

--- a/tests/system/test_system_async.py
+++ b/tests/system/test_system_async.py
@@ -48,7 +48,7 @@ from tests.system.test__helpers import (
 )
 
 
-RETRIES = retries.Retry(
+RETRIES = retries.AsyncRetry(
     initial=0.1,
     maximum=60.0,
     multiplier=1.3,

